lib/utf8.js

```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+module.exports.decode = function (input) {
+  if (Buffer.isBuffer(input) === false) {
```

Comment:
`utf8.encode` now returns an array instead of a buffer. No need for a third function and no performance loss.